src/celeritas/optical/surface/SimpleReflectionExecutor.hh

```diff
@@ -0,0 +1,93 @@
+//------------------------------- -*- C++ -*- -------------------------------//
+// Copyright Celeritas contributors: see top-level COPYRIGHT file for details
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file celeritas/optical/surface/SimpleReflectionExecutor.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include "corecel/Macros.hh"
+#include "corecel/math/ArrayOperators.hh"
+#include "celeritas/geo/GeoFwd.hh"
+#include "celeritas/optical/CoreTrackView.hh"
+
+#include "SurfaceInteraction.hh"
+#include "SurfacePhysicsData.hh"
+
+namespace celeritas
+{
+namespace optical
+{
+//---------------------------------------------------------------------------//
+/*!
+ */
+struct SimpleReflectionExecutor
+{
+    inline CELER_FUNCTION SurfaceInteraction operator()(CoreTrackView&) const;
```

Comment:
Nitpick: usually on structs we have data first, then functions. Is this a code guidance that we'd like to set?